### PR TITLE
added solution for spring-tests task

### PR DESCRIPTION
**Pull request by dmnichz**

### CODE REVIEW

File: src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.NoSuchElementException;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
```
Comment (Rommelua) [inline src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:12]: You have 3 methods in RoleDao but here only one is tested. Add more tests.
Same for UserDaoImplTest.

File: src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(1L, actual.getId());
+        Assertions.assertEquals("ADMIN", actual.getRoleName().name());
+    }
+
+    @Test
+    void getRoleByName_notOK() {
+        assertThrows(NoSuchElementException.class,
+                () -> roleDao.getRoleByName(Role.RoleName.USER.name()).get());
+    }
```
Comment (Rommelua) [inline src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:38]: I think we should add test case with passing null to getRoleByName()

File: src/main/java/mate/academy/controller/InjectController.java [hunk 3]
```diff
@@ -11,23 +15,38 @@
 @RequestMapping("/inject")
 public class InjectController {
```
Comment (Rommelua) [inline src/main/java/mate/academy/controller/InjectController.java:16]: Why did you change this class? The task is to create tests.
Same for RoleDao, RoleService.

File: src/test/java/mate/academy/dao/impl/UserDaoImplTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        user = new User();
+        user.setEmail("bob@i.ua");
+        user.setPassword("1234");
+        List<Role> roles = new ArrayList<>();
+        roles.add(new Role(Role.RoleName.ADMIN));
+        roles.add(new Role(Role.RoleName.USER));
+        user.setRoles(new HashSet<>(roles));
+        RoleDao roleDao = new RoleDaoImpl(getSessionFactory());
+        roleDao.save(roles.get(0));
+        roleDao.save(roles.get(1));
```
Comment (Rommelua) [inline src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:39]: Use roles.forEach()

File: src/test/java/mate/academy/dao/impl/UserDaoImplTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    @BeforeEach
+    void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        user = new User();
+        user.setEmail("bob@i.ua");
+        user.setPassword("1234");
+        List<Role> roles = new ArrayList<>();
+        roles.add(new Role(Role.RoleName.ADMIN));
+        roles.add(new Role(Role.RoleName.USER));
+        user.setRoles(new HashSet<>(roles));
```
Comment (Rommelua) [inline src/test/java/mate/academy/dao/impl/UserDaoImplTest.java:36]: Here and in the other places.
What is the reason of this List if you need Set?

File: src/test/java/mate/academy/security/AuthenticationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        roles = new ArrayList<>();
+        roles.add(new Role(Role.RoleName.ADMIN));
+        roles.add(new Role(Role.RoleName.USER));
+    }
+
+    @Test
+    void register_OK() {
+        Mockito.when(roleService.getRoleByName(any())).thenReturn(roles.get(1));
+        user.setRoles(Set.of(roles.get(1)));
+        Mockito.when(userService.save(any())).thenReturn(user);
```
Comment (Rommelua) [inline src/test/java/mate/academy/security/AuthenticationServiceImplTest.java:50]: Here and in the other places.
We should not use any() by default in our tests. Using any() is a potential mistake place. When you know exactly the argument that will be used inside your service method you should use it for mock objects.

File: src/test/java/mate/academy/service/impl/UserServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceImplTest {
+    private static final String EMAIL = "bob@i.ua";
+    private static final String PASSWORD = "1234";
+    private static final String ENCODED_PASSWORD = "%1234@";
+    private static final Role ADMIN = new Role(Role.RoleName.ADMIN);
```
Comment (Rommelua) [inline src/test/java/mate/academy/service/impl/UserServiceImplTest.java:22]: This approach is inconsistent with other test classes where you create role in setUp()

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void save_nullRole_notOK() {
+        try {
+            roleDao.save(null);
+        } catch (DataProcessingException e) {
+            Assertions.assertEquals("Can't create entity: null", e.getMessage());
+            return;
+        }
```
Comment (zachar-chorny) [inline src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:38]: use Assertions.assertThrows() instead of try/catch

File: src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Test
+    void getRoleByName_nullRole_notOK() {
+        try {
+            roleDao.getRoleByName(null);
+        } catch (DataProcessingException e) {
+            Assertions.assertEquals("Couldn't get role by role name: null", e.getMessage());
+            return;
+        }
+        Assertions.fail("Expected to receive DataProcessingException");
```
Comment (zachar-chorny) [inline src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java:59]: same. Fix it everywhere

File: src/test/java/mate/academy/security/AuthenticationServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService = new AuthenticationServiceImpl(userService,
```
Comment (zachar-chorny) [inline src/test/java/mate/academy/security/AuthenticationServiceImplTest.java:32]: Mock and initialize the current service better in the method annotated @BeforeAll instead of @BeforeEach. Fix this in all tests

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

**Comment by dmnichz** [comment]
Sorry for multiple requests, get confused with the `re-request review` button. Firstly it gets disappeared after clicking on it, now it's not.

**Review by zachar-chorny — APPROVED** [review]
